models/user.go

```diff
@@ -223,7 +223,7 @@ func (u *User) CanCreateRepo() bool {
 
 // CanCreateOrganization returns true if user can create organisation.
 func (u *User) CanCreateOrganization() bool {
-	return u.IsAdmin || u.AllowCreateOrganization
+	return u.IsAdmin || (u.AllowCreateOrganization && !setting.Admin.DisableRegularOrgCreation)
```

Comment:
And I think maybe you could let this option disabled and default value is `setting.Admin.DisableRegularOrgCreation` when `setting.Admin.DisableRegularOrgCreation` is false and change  https://github.com/go-gitea/gitea/blob/master/models/user.go#L655 to the UI form's value?